server/internal/app/auth_client.go

```diff
@@ -126,37 +127,36 @@ func attachIntegrationOperator(ctx context.Context, req *http.Request, cfg *Serv
 	return ctx, nil
 }
 
-func PublicAPIAuthMiddleware(cfg *ServerConfig) echo.MiddlewareFunc {
+func publicAPIAuthMiddleware(cfg *ServerConfig) echo.MiddlewareFunc {
 	return func(next echo.HandlerFunc) echo.HandlerFunc {
 		return func(c echo.Context) error {
-			// TODO: support limit publication scope
-
-			// req := c.Request()
-			// ctx := req.Context()
-			// if token := req.Header.Get("Reearth-Token"); token != "" {
-			// 	ws, err := cfg.Repos.Project.FindByPublicAPIToken(ctx, token)
-			// 	if err != nil {
-			// 		if errors.Is(err, rerror.ErrNotFound) {
-			// 			return c.JSON(http.StatusUnauthorized, map[string]string{"error": "invalid token"})
-			// 		}
-			// 		return err
-			// 	}
-
-			// 	if !ws.IsPublic() {
-			// 		return c.JSON(http.StatusUnauthorized, map[string]string{"error": "invalid token"})
-			// 	}
-
-			// 	c.SetRequest(req.WithContext(adapter.AttachOperator(ctx, &usecase.Operator{
-			// 		PublicAPIProject: ws.ID(),
-			// 	})))
-			// }
+			req := c.Request()
+			ctx := req.Context()
+			if token := getToken(req); token != "" {
+				p, err := cfg.Repos.Project.FindByPublicAPIToken(ctx, token)
+				if err != nil {
+					if errors.Is(err, rerror.ErrNotFound) {
+						return c.JSON(http.StatusUnauthorized, map[string]string{"error": "invalid token"})
+					}
+					return err
+				}
+
+				if p.Publication().Scope() == project.PublicationScopePrivate {
```

Comment:
possible null exception 
```suggestion
				if p.Publication() == nil || p.Publication().Scope() == project.PublicationScopePrivate {
```